pydantic_evals/pydantic_evals/evaluators/llm_as_a_judge.py

```diff
@@ -100,6 +102,62 @@ async def judge_output(
 )
 
 
+def _build_prompt_with_input(
+    inputs: Any,
+    output: Any,
+    rubric: str,
+    expected_output: Any | None = None,
+) -> str | Sequence[UserContent]:
+    """Build a prompt that includes input, output, and rubric.
+
+    If inputs is a string, returns a formatted string.
+    If inputs is not a string (e.g., BinaryContent), returns a list of UserContent.
+    """
+    if isinstance(inputs, str):
+        # For string inputs, use simple string formatting
+        if expected_output is not None:
+            return dedent(f"""
+            <Input>
+            {inputs}
+            </Input>
+            <ExpectedOutput>
+            {_stringify(expected_output)}
+            </ExpectedOutput>
+            <Output>
+            {_stringify(output)}
+            </Output>
+            <Rubric>
+            {rubric}
+            </Rubric>
+            """)
+        else:
+            return dedent(f"""
+            <Input>
+            {inputs}
+            </Input>
+            <Output>
+            {_stringify(output)}
+            </Output>
+            <Rubric>
+            {rubric}
+            </Rubric>
+            """)
```

Comment:
Similar to below, can we have build a list of strings and conditionally add the item for `ExpectedOutput`, and `join` them all at the end? That'll reduce the duplication a bit.

We can likely also merge the 2 `isinstance(inputs, str)` branches, and only `join` if `isinstance(inputs, str)`, and return the entire list otherwise

Reply:
I've refactored it to reduce duplication